solidity/random-beacon/contracts/libraries/Relay.sol

```diff
@@ -133,23 +127,36 @@ library Relay {
             "Invalid entry"
         );
 
-        // If the soft timeout has been exceeded apply stake slashing for
-        // all group members. Note that `getSlashingFactor` returns the
-        // factor multiplied by 1e18 to avoid precision loss. In that case
-        // the final result needs to be divided by 1e18.
-        slashingAmount =
-            (getSlashingFactor(self, dkgGroupSize) *
-                self.relayEntrySubmissionFailureSlashingAmount) /
-            1e18;
-
         emit RelayEntrySubmitted(self.currentRequestID, msg.sender, entry);
 
         self.previousEntry = AltBn128.g1Unmarshal(entry);
         self.currentRequestID = 0;
         self.currentRequestGroupID = 0;
         self.currentRequestStartBlock = 0;
+    }
+
+    /// @notice Calculates the slashing amount for all group members.
+    /// @dev Must be used when a soft timeout was hit.
+    /// @return Amount by which group members should be slashed
+    ///         in case the relay entry was submitted after the soft timeout.
+    function calculateSlashingAmount(Data storage self)
```

Comment:
Somehow I don't feel comfortable changing calculation logic from `getSlashingFactor` to `calculateSlashingAmount`. Would it be possible to leave the old `getSlashingFactor` and call it in  `calculateSlashingAmount`?

Reply:
Do you have a specific concern? I've tried to maintain the core logic. If we breakdown the pieces for slashing amount calculation this is what we end up with:
```
slashingAmount = (slashingFactor * relayEntrySubmissionFailureSlashingAmount) / 1e18
softTimeoutBlock = self.currentRequestStartBlock + (groupSize * relayEntrySubmissionEligibilityDelay)
submissionDelay = block.number - softTimeoutBlock;
slashingFactor = (submissionDelay * 1e18) / relayEntryHardTimeout

slashingAmount = ((submissionDelay * 1e18) / relayEntryHardTimeout) * relayEntrySubmissionFailureSlashingAmount / 1e18
slashingAmount = submissionDelay * relayEntrySubmissionFailureSlashingAmount / relayEntryHardTimeout 
```
so `slashingAmount` is what we have in `calculateSlashingAmount()`. Also:
- `slashingFactor > 1e18 ? 1e18 : slashingFactor` was replaced with `if (submissionDelay > self.relayEntryHardTimeout) { .. } `. It guards from cases when we pass the hard timeout and we won't exceed the slashing max amount.
- `if (block.number > softTimeoutBlock) { .. }` can be added, but I think that would be a duplicate check, see [this](https://github.com/keep-network/keep-core/pull/2772#discussion_r776777137) comment.